### PR TITLE
fix: loader type

### PR DESCRIPTION
**Pull request by wtlin1228**
# Overview

Fix loader type for the case: `loaderQrl` returns `fail(...)` without any validator.

Related issue: https://github.com/BuilderIO/qwik/issues/3979

# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [x] Docs / tests / types / typos

# Description

I only do `StrictUnion` when the `loaderQrl` returns `fail(...)` to keep the most simple case's type the same.

```ts
// Loader<number>
export const useLoader1 = routeLoader$(() => 123); 

// Loader<{
//     foo: number;
// }>
export const useLoader2 = routeLoader$(() => ({ foo: 123 })); 
```

And for those returning `fail(...)`:

```ts
// Loader<{
//     loaderFailedReason?: string | undefined;
//     failed?: true | undefined;
//     pet?: undefined;
// } | {
//     pet?: string | undefined;
//     failed?: undefined;
//     loaderFailedReason?: undefined;
// }>
export const useRandomFailedLoader = routeLoader$(({ fail }) => {
  if (Math.random() > 0.5) {
    return fail(500, {
      loaderFailedReason: "Reach Limit",
    });
  }

  return {
    pet: "guinea pig",
  };
});
```

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/qwik-city/runtime/src/types.ts [hunk 1]
```diff
@@ -729,12 +729,14 @@ export interface ActionStore<RETURN, INPUT, OPTIONAL extends boolean = true> {
   >;
 }
 
+type Failed = {
+  failed: true;
+};
+
```
Comment (wtlin1228) [inline packages/qwik-city/runtime/src/types.ts:735]: For reviewers: 

Should this type be exported?

**Review by mhevery — COMMENTED** [review]
Thank you for the contribution!

I see you have an issue with reproduction, but I don't see a test. Did you forget to check it in?

**Comment by wtlin1228** [comment]
> Thank you for the contribution!
> 
> I see you have an issue with reproduction, but I don't see a test. Did you forget to check it in?

Hi @mhevery, test is `starters/apps/qwikcity-test/src/routes/(common)/loaders/issue3979/index.tsx`.

**Review by NexonTreeHouse — APPROVED** [review]

**Review by mhevery — APPROVED** [review]
Thank you for your contribution. ❤️

File: packages/qwik-city/runtime/src/types.ts (hunk 1 above)
Comment (mhevery) [reply packages/qwik-city/runtime/src/types.ts:735]: I think it is fine not being exported.

**Pull request merged**